packages/@textlint/kernel/src/task/fixer-task.ts

```diff
@@ -63,6 +63,7 @@ export default class TextLintCoreTask extends CoreTask {
             report,
             configBaseDir: this.configBaseDir
         });
+        Object.freeze(ruleContext);
```

Comment:
You can create a factory function like `createRuleContext`.

```ts
export const createFreezedRuleContext = (args: RuleContextArgs) => {
   return Object.freeze(new RuleContext(arg));
}
// FIXME: default export => named export
export default class RuleContext implements BaseRuleContext {
}
```

This pattern is type-safe and I like it.

```ts
const obj = { key: "str" }
const freezedObject = Object.freeze(obj);
freezedObject.key = "test"; // freezedObject is read only
```

- https://agentcooper.github.io/typescript-play/?noImplicitAny=false&target=2#code/MYewdgzgLgBCBGArGBeGBvGBrApgTwC4YAiaAJ2JgF8AoUSWAMzJxwC8cATAeSR2FhpeiflAB0zVhwAUCRAEoA3DUnsuw0WNx5UJKDmjFFQA

Reply:
I'm done. How do you feel about my changes?